crates/core/src/error/collector.rs

```diff
@@ -0,0 +1,289 @@
+use derive_more::From;
+
+use crate::{error::RusticErrorKind, RusticError};
+use log::{error, info, warn};
+
+/// A rustic issue result
+///
+/// rustic issue results are used to return a result along with possible issues.
+#[derive(Debug)]
+pub enum RusticIssueResult<T> {
+    /// For when we return a result along with possible issues
+    Ok(T, Option<Vec<RusticIssue>>),
+
+    /// For when we abort and return only the errors that occurred
+    Err(Vec<RusticError>),
+}
+
+/// A rustic issue
+///
+/// An issue is a message that can be logged to the user.
+#[derive(Debug)]
+pub enum RusticIssue {
+    /// An error issue, indicating that something went wrong irrecoverably
+    Error(RusticError),
+
+    /// A warning issue, indicating that something might be wrong
+    Warning(RusticWarning),
+
+    /// An info issue, indicating additional information
+    Info(RusticInfo),
+}
+
+impl RusticIssue {
+    pub fn new_error(error: RusticErrorKind) -> Self {
+        Self::Error(error.into())
+    }
+
+    pub fn new_warning(message: &str) -> Self {
+        Self::Warning(message.into())
+    }
+
+    pub fn new_info(message: &str) -> Self {
+        Self::Info(message.into())
+    }
+
+    pub fn log(&self) {
+        match self {
+            Self::Error(error) => error!("{}", error),
+            Self::Warning(warning) => warn!("{}", warning.0),
+            Self::Info(info) => info!("{}", info.0),
+        }
+    }
+}
+
+/// A rustic warning message
+///
+/// Warning messages are used to indicate that something might be wrong.
+#[derive(Debug, Clone, From)]
+pub struct RusticWarning(String);
```

Comment:
I don't like having `String` here. IMO we should just have `RusticError`

Reply:
Same as above, this here is a `warning`, `RusticError` should be something that we return to the user, in case we can't recover from something. This is pub here, but I'm not sure if we ever should make RusticWarning, RusticInfo part of our public API. Users of `rustic_core`, also `rustic-rs` should never have to deal with that. They will only get `errors` from us. I think 🤔 If it is recoverable, then we should recover from it, not the users.